### PR TITLE
Add builder prune --all and --force flag support

### PR DESCRIPTION
**Pull request by austinvazquez**
### Description

This change adds support for builder prune `--all` and `--force`. A refactor
of previous tests which used builder prune functionality was required to
handle the new confirmation prompt. A test utility was added to handle
this for reuse by current and future tests.

### Testing

From `containerd/nerdctl`:
```
nerdctl builder build .
nerdctl builder prune --all --force
```

### Next steps
1. Test assertions are weak without a `builder du` implementation

### CODE REVIEW

File: cmd/nerdctl/builder.go [hunk 1]
```diff
@@ -58,29 +58,83 @@ func newBuilderPruneCommand() *cobra.Command {
...
 	}
-	buildkitHost, err := getBuildkitHost(cmd, globalOptions.Namespace)
+
+	if !options.Force {
+		var (
+			confirm string
+			msg     string
+		)
+
+		if !options.All {
```
Comment (pendo324) [inline cmd/nerdctl/builder.go:79]: nit: might be slightly easier to read if you reverse these conditionals since we need to handle both cases anyway

**Review by pendo324 — APPROVED** [review]

File: pkg/api/types/builder_types.go [hunk 2]
```diff
@@ -82,4 +82,6 @@ type BuilderPruneOptions struct {
 	BuildKitHost string
 	// All will remove all unused images and all build cache, not just dangling ones
 	All bool
+	// Force will not prompt for confirmation.
+	Force bool
```
Comment (AkihiroSuda) [inline pkg/api/types/builder_types.go:86]: Where is this consumed?

Comment (austinvazquez) [reply pkg/api/types/builder_types.go:86]: Looks like it ended up being used on the CLI side only, so not much reason to pass it to the package backend.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**